egs/wsj/s5/steps/dict/learn_lexicon.sh

```diff
@@ -0,0 +1,402 @@
+    $ref_dict/silence_phones.txt $dir/lats2/pron_stats.txt $dir/train_counts.txt $dir/ref_lexicon.txt \
+    $dir/g2p_pron_candidates_pruned.txt $dir/phone_decoding_generated_prons_pruned.txt \
+    $dir/lats2/pron_posteriors.temp $dir/lats2/out_of_ref_vocab_prons_learned.txt $dir/lats2/ref_lexicon_edits.txt
+
+  # We reformat the pron_posterior file and add some comments.
+  paste <(cat $dir/lats2/pron_posteriors.temp | cut -d' ' -f1-3 | column -t) \
+    <(cat $dir/lats2/pron_posteriors.temp | cut -d' ' -f4-) | sort -nr -k1,3 | \
+    cat <( echo ';; <word> <source: R(eference)/G(2P)/P(hone-decoding)> <posterior> <pronunciation>') -  \
+    > $dir/lats2/pron_posteriors.txt
+  rm $dir/pron_posteriors.temp 2>/dev/null
+
+  # Remove some stuff that takes up space and is unlikely to be useful later on.
+  if $cleanup; then
+    rm -r $dir/lats2/{fsts*,lat*} 2>/dev/null
+  fi
+fi
+
+if [ $stage -le 7 ]; then
+  echo "$0: Expand the learned lexicon further to cover words in target vocab that are."
+  echo "  ... not seen in acoustic training data."
+  mkdir -p $dest_dict
+  cp $ref_dict/{extra_questions.txt,optional_silence.txt,nonsilence_phones.txt,silence_phones.txt} \
+    $dest_dict  2>/dev/null
+  rm $dest_dict/lexiconp.txt 2>/dev/null
+  # Get the list of oov (w.r.t. ref vocab) without acoustic evidence, which are in the
+  # target vocab. We'll just assign to them pronunciations from g2p_pron_candidates, if any.
+  cat $dir/lats2/out_of_ref_vocab_prons_learned.txt $dir/ref_lexicon.txt | \
+    awk 'NR==FNR{a[$1] = 1; next} !($1 in a)' - \
+    $dir/target_vocab.txt | sort | uniq > $dir/oov_no_acoustics.txt
+
+  awk 'NR==FNR{a[$1] = 1; next} ($1 in a)' $dir/oov_no_acoustics.txt \
+    $dir/g2p_pron_candidates > $dir/g2p_prons_for_oov_no_acoustics.txt
+  
+  cat $dir/g2p_prons_for_oov_no_acoustics.txt $dir/lats2/out_of_ref_vocab_prons_learned.txt \
+    $dir/ref_lexicon.txt | tr -s '\t' ' ' | sort | uniq > $dest_dict/lexicon.temp
+
+  awk 'NR==FNR{a[$1] = 1; next} ($1 in a)' $dir/target_vocab.txt \
+    $dest_dict/lexicon.temp | sort | uniq > $dest_dict/lexicon.nosil
+
+  cat $dir/non_scored_entries $dest_dict/lexicon.nosil | sort | uniq >$dest_dict/lexicon0.txt
+fi
+
+if [ $stage -le 8 ]; then
+  if [ $apply_edits ]; then
+    echo "$0: Apply the ref_lexicon_edits file to the reference lexicon."
+    cp $dir/lats2/ref_lexicon_edits.txt $dest_dict/lexicon_edits.txt 2>/dev/null
+    steps/dict/apply_lexicon_edits.py $dest_dict/lexicon0.txt $dir/lats2/ref_lexicon_edits.txt - | \
+      sort -u > $dest_dict/lexicon.txt
+  else
+    echo "$0: The user requested not to apply the ref_lexicon_edits file. So we just exit at this point."
+    echo "  ... and leave it to the user to inspect/modify the edits file and then apply it on"
+    echo "  ... $dest_dict/lexicon0.txt to produce the final learned lexicon $dest_dict/lexicon.txt."
```

Comment:
Please print out the exact command line the user must run.
Also, what's wrong with always applying the edits, but also always printing a message saying that you can modify the edits and then reapply them with such-and-such a command line?